bitcoin/wallet.go

```diff
@@ -213,6 +215,34 @@ func (w *BitcoinWallet) Transactions() ([]wi.Txn, error) {
 
 func (w *BitcoinWallet) GetTransaction(txid chainhash.Hash) (wi.Txn, error) {
 	txn, err := w.db.Txns().Get(txid)
+	if err == nil {
+		tx := wire.NewMsgTx(1)
+		rbuf := bytes.NewReader(txn.Bytes)
+		err := tx.BtcDecode(rbuf, wire.ProtocolVersion, wire.WitnessEncoding)
+		if err != nil {
+			return txn, err
+		}
+		outs := []wi.TransactionOutput{}
+		for i, out := range tx.TxOut {
+			var addr btc.Address
+			_, addrs, _, err := txscript.ExtractPkScriptAddrs(out.PkScript, w.params)
+			if err != nil {
+				log.Printf("error extracting address from txn pkscript: %v\n", err)
+			}
+			if len(addrs) == 0 {
+				addr = nil
+			} else {
+				addr = addrs[0]
+			}
+			tout := wi.TransactionOutput{
+				Address: addr,
+				Value:   out.Value,
+				Index:   uint32(i),
+			}
+			outs = append(outs, tout)
+		}
+		txn.Outputs = outs
+	}
```

Comment:
This whole function should be reduced to `util.ReadOutputs(txn.Bytes) []wi.TransactionOutput` and shared in each implementation. None of them appear to be specific to the coin and seem shareable.

Reply:
Well it does need to be coin specific, but this does bring up a point that the other ones are using the wrong lib.